### PR TITLE
Context menu codelab first draft, including starter and completed code.

### PR DESCRIPTION
**Pull request by rachel-fenichel**
Adds:
- A new folder in the codelabs directory, with the codelab text and a few images.
- A new folder in the examples directory, with two subfolders:
  - starter_code: a very basic page with a Blockly workspace.
  - complete_code: the final version of the code written in the codelab.

Right now I'm looking for feedback on:
- Structure
- Anything I'm missing
- Anything I'm misstating

Remaining work:
- Clean up and comment all of the finished code, to make it readable as a standalone example.
- Make sure code in the codelab matches code in `complete_code` after cleanup.
- Add images as needed
- Actually run with claat and make sure everything compiles
- Update the readme in the examples directory to be more useful
- Final edits

### CODE REVIEW

File: examples/context-menu-codelab/complete-code/index.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    displayText: function(scope) {
+      if (scope.block.type.startsWith('text')) {
+        return 'Text block';
+      } else if (scope.block.type.startsWith('controls')) {
+        return 'Controls block';
+      } else {
+        return 'Some other block';
+      }
+    },
+    preconditionFn: function (scope) {
```
Comment (maribethb) [inline examples/context-menu-codelab/complete-code/index.js:108]: nit: no space after `function` here and below

**Review by maribethb — APPROVED** [review]
Looks good! Does the codelab tool automatically turn every h2 into a new page?

**Review by maribethb — COMMENTED** [review]
Also you may or may not want to add something about how it's a singleton so they should not instantiate a new ContextMenuRegistry object and only use `Blockly.ContextMenuRegistry.registry` to access. I tried to make it clear in the code but it might be a good thing to note here as well.

**Comment by rachel-fenichel** [comment]
Updated:
- Added a note about not instantiating the registry object
- Fixed spacing
- Added comments in the completed code
- Added a link to the codelab in the README for the completed code
- Added images
- Ran `claat export` to check the formatting of the codelab after passing through the tool
- Changed most weights to 100 to make sure they are at the bottom of the menu
- Removed extra spaces
- Added a link to the sample from gh-pages

And yes, the tool automatically makes a new page for each h2.

**Pull request merged**